### DISCUSSION TITLE
Alphabetical sorting of dictionaries

### DISCUSSION THREAD

**Discussion by ichnilatis-gr** [koreader/koreader#8819] [Q&A]
I had the following notification displayed:

```
The user dictionary
./settings/user-Greek.hyph
is not alphabetically sorted.

It has been disabled.
```

What can I do to enable it again?
Thanks!

**Comment by ichnilatis-gr** [comment]
Let me put it another way:
How can I sort a dictionary alphabetically?

**Reply by Frenzie** [reply]
Pinging @zwim

**Reply by ghost** [reply]
If you use linux this is easy:
1. Copy the file `user-Greek.hyph` to your PC (rename it to `user-Greek.hyph.orig` and make a backup, just to be safe)
2. enter on the command line `sort user-Greek.hyph.orig > user-Greek.hyph`
3. Copy the new `user-Greek.hyph` file back to the settings folder.

Alternatively you can try the brand new terminal emulator (depending if your device has `sort`)
1. Open terminal
2. `cd settings`
3. `mv user-Greek.hyph user-Greek.hyph.orig`
4. `sort user-Greek.hyph.orig > user-Greek.hyph`

**Reply by ghost** [reply]
@ichnilatis-gr : I'm curious: How did you get a non alphabetically sorted dictionary? I hope not from KOReader but from somewhere else.

**Reply by ichnilatis-gr** [reply]
This dictionary stores my different hyphenation choices. One time, when I tapped on a word to select then the hyphenation option, that popup appeared.

**Reply by ichnilatis-gr** [reply]
> * `cd settings`
> 
> * `mv user-Greek.hyph user-Greek.hyph.orig`
> 
> * `sort user-Greek.hyph.orig > user-Greek.hyph`

Will I use the terminal of KOreader and write these lines in it? Do I understand correctly?

**Reply by Frenzie** [reply]
Yup, exactly. I forget which device you have but on Kobo that should work.

**Reply by ichnilatis-gr** [reply]
PB 628

**Reply by ichnilatis-gr** [reply]
It seems that I can't use terminal emulator. Crashes are happening.

**Reply by ichnilatis-gr** [reply]
It says: "Open terminal session (not running)"

**Reply by ichnilatis-gr** [reply]
What else can I do?

**Reply by ghost** [reply]
You can open an issue on KOReader "User hyphenation file not alphabetically sorted" and upload your misbehaving `user-Greek.hyph` there. Then I can analyze the problem and maybe fix it (even if I don't speak greek).

**Reply by ghost** [reply]
> It says: "Open terminal session (not running)"

What happens if you tap on that menu item?

**Reply by ichnilatis-gr** [reply]
It crashes.

**Reply by ghost** [reply]
That's bad. You can open an issue for that too, please include the crash.log.

**Reply by ichnilatis-gr** [reply]
Done.

**Reply by ghost** [reply]
I will look into both on WE.
